File: java/java-sdk/src/main/java/io/kestra/sdk/model/ApiPolicyMutationAction.java
/*
 * Kestra EE
 * All API operations, except for Superadmin-only endpoints, require a tenant identifier in the HTTP path.<br/> Endpoints designated as Superadmin-only are not tenant-scoped.
 */


package io.kestra.sdk.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Gets or Sets ApiPolicyMutationAction
 */
public enum ApiPolicyMutationAction {
  
  ADDED("ADDED"),
  
  REPLACED("REPLACED"),
  
  REMOVED("REMOVED"),
  
  UNKNOWN_DEFAULT_OPEN_API("unknown_default_open_api");

  private String value;

  ApiPolicyMutationAction(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static ApiPolicyMutationAction fromValue(String value) {
    for (ApiPolicyMutationAction b : ApiPolicyMutationAction.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    return UNKNOWN_DEFAULT_OPEN_API;
  }
}

File: python/python-sdk/kestrapy/models/api_policy_mutation_action.py
# coding: utf-8

"""
    Kestra EE

    All API operations, except for Superadmin-only endpoints, require a tenant identifier in the HTTP path.<br/> Endpoints designated as Superadmin-only are not tenant-scoped.
"""  # noqa: E501


from __future__ import annotations
import json
from enum import Enum
from typing_extensions import Self


class ApiPolicyMutationAction(str, Enum):
    """
    ApiPolicyMutationAction
    """

    """
    allowed enum values
    """
    ADDED = 'ADDED'
    REPLACED = 'REPLACED'
    REMOVED = 'REMOVED'

    @classmethod
    def from_json(cls, json_str: str) -> Self:
        """Create an instance of ApiPolicyMutationAction from a JSON string"""
        return cls(json.loads(json_str))
